ush/python/pygfs/task/aero_prepobs.py

```diff
@@ -0,0 +1,237 @@
+                        matching_files.append(os.path.join(self.task_config.data_dir, file))
+            logger.info("Found %d matching files.", len(matching_files))
+        except FileNotFoundError:
+            logger.error("The specified file/directory does not exist.")
+            raise
+        return matching_files
+
+    @logit(logger)
+    def copy_obs(self, inputfiles) -> Dict[str, Any]:
+        """
+        Copy the raw obs files to $DATA/obs.
+        """
+        copylist = []
+        destlist = []
+        for filename in inputfiles:
+            basename = os.path.basename(filename)
+            dest = os.path.join(self.task_config.DATA_OBS, basename)
+            copylist.append([filename, dest])
+            destlist.append(dest)
+        FileHandler({'copy': copylist}).sync()
+
+        return destlist
+
+    @logit(logger)
+    def get_obsproc_config(self, sensor) -> Dict[str, Any]:
+        """
+        Compile a dictionary of obs proc configuration from OBSPROCYAML template file
+        Parameters
+        ----------
+        Returns
+        ----------
+        obsproc_config : Dict
+            a dictionary containing the fully rendered obs proc yaml configuration
+        """
+        self.task_config.sensor = sensor
+        # generate JEDI YAML file
+        logger.info(f"Generate gdas_obsprovider2ioda YAML config: {self.task_config.OBSPROCYAML}")
+        prepaero_config = parse_j2yaml(self.task_config.OBSPROCYAML, self.task_config)
+
+        return prepaero_config
+
+    @logit(logger)
+    def link_obsconvexe(self) -> None:
+        """
+        This method links the gdas executable to the run directory
+        Parameters
+        ----------
+        Task: GDAS task
+        Returns
+        ----------
+        None
+        """
+        exe_src = self.task_config.OBSPROCEXE
+
+        logger.info(f"Link executable {exe_src} to DATA/")
+        exe_dest = os.path.join(self.task_config.DATA, os.path.basename(exe_src))
+        if os.path.exists(exe_dest):
+            rm_p(exe_dest)
+        os.symlink(exe_src, exe_dest)
+
+        return
+
+    @logit(logger)
+    def runConverter(self) -> None:
+        """
+        Run the IODA converter gdas_obsprovider2ioda.x
+        """
+        chdir(self.task_config.DATA)
+        for prepaero_yaml in self.task_config.prepaero_yaml:
+            exec_cmd = Executable(self.task_config.APRUN_PREPOBSAERO)
+            exec_name = os.path.join(self.task_config.DATA, 'gdas_obsprovider2ioda.x')
+            exec_cmd.add_default_arg(exec_name)
+            exec_cmd.add_default_arg(prepaero_yaml)
+
+            try:
+                logger.debug(f"Executing {exec_cmd}")
+                exec_cmd()
+            except OSError:
+                raise OSError(f"Failed to execute {exec_cmd}")
+            except Exception:
+                raise WorkflowException(f"An error occured during execution of {exec_cmd}")
```

Comment:
The `add_default_arg` `Executable` method sets arguments that should be used for all calls.  Other arguments can be sent at call time.  Instead of constructing an `Executable` object each time and appending arguments with `add_default_arg`, try creating just one `Executable`:
```suggestion
        exec_cmd = Executable(self.task_config.APRUN_PREPOBSAERO)
        exec_name = os.path.join(self.task_config.DATA, 'gdas_obsprovider2ioda.x')
        exec_cmd.add_default_arg(exec_name)

        for prepaero_yaml in self.task_config.prepaero_yaml:
            try:
                logger.debug(f"Executing {exec_cmd} on {prepaero_yaml}")
                exec_cmd(f"{prepaero_yaml}")
            except OSError:
                raise OSError(f"Failed to execute {exec_cmd} on {prepaero_yaml}")
            except Exception:
                raise WorkflowException(f"An error occured during execution of {exec_cmd} on {prepaero_yaml}")